enrolments/serializers.py

```diff
@@ -79,3 +85,44 @@ class Meta:
             "enrolled_class",
             "status",
         ]
+
+
+class EnrolmentCreateSerializer(serializers.ModelSerializer):
+    family = FamilyDetailSerializer()
+    preferred_class = serializers.PrimaryKeyRelatedField(
+        allow_null=True, queryset=Class.objects.all()
+    )
+    session = serializers.PrimaryKeyRelatedField(queryset=Session.objects.all())
+
+    class Meta:
+        model = Enrolment
+        fields = [
+            "family",
+            "session",
+            "preferred_class",
+        ]
+
+    def create(self, validated_data):
+        family = FamilyDetailSerializer.create(self, validated_data["family"])
+        students = Student.objects.filter(family=family)
```

Comment:
you could also try using `values_list` (like [here](https://github.com/uwblueprint/project-read-backend/blob/0e49b99bf21b1aca497d474ee096b6beb69cc902/enrolments/validators.py#L80)) to avoid the for loop later